common/autoconf/jdk-options.m4

```diff
@@ -568,7 +569,9 @@ AC_DEFUN_ONCE([JDKOPT_SETUP_JDK_VERSION_NUMBERS],
     AC_MSG_ERROR([--with-vendor-url must have a value])
   elif [ ! [[ $with_vendor_url =~ ^[[:print:]]*$ ]] ]; then
     AC_MSG_ERROR([--with-vendor-url contains non-printing characters: $with_vendor_url])
-  else
+  elif test "x$with_vendor_url" != x; then
```

Comment:
This doesn't handle the use of `--without-vendor-url`.

Reply:
Such a thing is not supported afaik.

Reply:
Here's an excerpt from `bash configure --help`:
```
Optional Packages:
  --with-PACKAGE[=ARG]    use PACKAGE [ARG=yes]
  --without-PACKAGE       do not use PACKAGE (same as --with-PACKAGE=no)
```
(in this case, 'PACKAGE' is 'vendor-url').

Reply:
Setting `--with-vendor-url=yes` is an error. I think this is correct, and duplicates how it works for jdk11+.

Reply:
The last line I quoted was the one that is relevant here; the code should be:
```
elif test "x$with_vendor_url" != x -a "x$with_vendor_url" != xno ; then
```

Reply:
Actually, that's not what a user might expect; if they say `--without-vendor-url`, I think they should get no URL, not the default, so we want something like this:
```
elif test "x$with_vendor_url" = xno ; then
  VENDOR_URL=
elif test "x$with_vendor_url" != x ; then
  VENDOR_URL="$with_vendor_url"
```

Reply:
That's no how any of the other settings work, or how jdk11+ works.

Reply:
See:
* https://github.com/ibmruntimes/openj9-openjdk-jdk8/blob/openj9/common/autoconf/basics.m4#L481
* https://github.com/ibmruntimes/openj9-openjdk-jdk8/blob/openj9/common/autoconf/toolchain.m4#L857

Reply:
By "other settings" I mean `--with-vendor-bug-url`, `--with-vendor-vm-bug-url`, `--with-vendor-name` (possibly more). I don't think we should have `--with_vendor_url` work differently from these other options or from jdk11+.

Reply:
Running configure without `--with-vendor-url` today yields
```
VENDOR_URL:=
```
in `spec.gmk`; with this change, there is no way to achieve that.
With my suggestion, `--without-vendor-url` would leave it blank.

Reply:
jdk11+ doesn't allow setting VENDOR_URL blank either. I don't think we should support having it blank, it would affect the system properties set by OpenJ9.

Reply:
This excerpt from jdk11's `spec.gmk.in` would seem to suggest otherwise
```
# Only export VENDOR_URL, VENDOR_URL_BUG and VENDOR_VM_URL_BUG to the build if
# they are not empty. Otherwise, default values which are defined in the sources
# will be used.
ifneq ($(VENDOR_URL),)
  VERSION_CFLAGS += -DVENDOR_URL='"$(VENDOR_URL)"'
endif
```
but it appears you're right, `VENDOR_URL` can't be blank there.